replica/table.cc

```diff
@@ -1372,6 +1372,9 @@ compaction_group::update_main_sstable_list_on_compaction_completion(sstables::co
 future<>
 table::compact_all_sstables(std::optional<tasks::task_info> info) {
     co_await flush();
+    // Forces off-strategy before major, so sstables previously sitting on maintenance set will be included
+    // in the compaction's input set, to provide same semantics as before maintenance set came into existence.
+    co_await perform_offstrategy_compaction(info);
```

Comment:
I have a question: If I understand correctly, this code first perform an "off-strategy compaction" and waits for it to finish, and only then a major compaction. But why? Can't we just tell the major compaction to include all sstables, and avoid having to do two compactions one after the other, which (unless I'm misunderstanding something) just increases write amplification?

Reply:
I am working with existing building blocks. Major doesn't operate today across sets. If we think that maintenace is much smaller than main, then it's not a problem that we compact maintenance first, then main set. Had it been the other way around, then it'd be terrible. Like compacting 1TB(main) first, then later 1TB(main)+10G(maintenace). But given maintenance happens first, so write amplification is only the size of maintenace. Perhaps add a FIXME so we can revisit it later if needed?

Reply:
Yes, sounds good to me. Even if we don't bother to fix it any time soon, I think it will be helpful to remember you didn't do this in two steps deliberately, and it's just an implementation detail.
I thought maybe there might have been some deliberate reason why you did it this way, perhaps "off strategy compaction" could do something faster somehow, but it didn't sound right.